### PR TITLE
Add documentation for SignalR Service internals

### PR DESCRIPTION
**Pull request by xscript**

### CODE REVIEW

File: docs/internal.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+- [Transport Data between Client and Server](#transport)
+
+Azure SignalR Service is built on top of ASP.NET Core SignalR framework.
+You can start from a local ASP.NET Core SignalR application, then seamlessly move to Azure SignalR Service with a few lines of code change.
+
+The following diagram shows the typical architecture when you use Azure SignalR Service with your own application server.
+Behavior of clients and servers will be different from self-hosting ASP.NET Core SignalR.
+More details are explained in below sections.
+
+![Architecture](./images/arch.png)
```
Comment (davidfowl) [inline docs/internal.md]: This diagram should show a 2 way connection between the client and service

Comment (xscript) [reply docs/internal.md]: Sure. The arrow is hidden by the client icon. Let me update it.

File: docs/internal.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+## Server Connections
+
+When self-hosting ASP.NET Core SignalR, application server will listen for client connections and serve them directly.
+
+With SignalR Service, application server will not accept SignalR client connections anymore.
```
Comment (davidfowl) [inline docs/internal.md]: "the application server"

Comment (davidfowl) [reply docs/internal.md]: Lets change this to say:

"With SignalR Service, the application server is no longer responsible for accepting persistent client connections."

File: docs/internal.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+A `negotiate` endpoint will be exposed by Azure SignalR Service SDK for each hub.
+It will respond to clients' negotiate requests and redirect them to SignalR Service.
+Eventually, all clients will be connected to SignalR Service.
+Read more about it in [Client Connections](#client-connections) section.
+
+Right after application server is started, a few WebSocket connections are initiated to SignalR Service.
```
Comment (davidfowl) [inline docs/internal.md]: Remove "Right"

Comment (davidfowl) [reply docs/internal.md]: Change WebSocket to lowercase.

File: docs/internal.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+By default the connection count is 5 and it is configurable.
```
Comment (davidfowl) [inline docs/internal.md]: After application server is started, the Azure SignalR SDK will open 5 WebSocket connections to the SignalR Service. The number of used connections is configurable.

File: docs/internal.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+Apparently, client count will be far more than the connnection count between application server and SignalR Service.
```
Comment (davidfowl) [inline docs/internal.md]: Why Apparently? You can remove this sentence completely.

File: docs/internal.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+It is a two-step process to establish persistent connections between clients and SignalR Service.
+
+1. Client sends a negotiate request to application server.
+With Azure SignalR Service SDK, application server will return a redirect negotiation response with SignalR Service's URL and access token.
+Typical redirect response looks like as following:
+
+    ```json
+    {
+        "url":"https://test.service.signalr.net:5001/client/?hub=chat&...",
+        "accessToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYmYiOjE1MzY2NTYzMjYsImV4cCI6MTUzNjY1OTkyNiwiaWF0IjoxNTM2NjU2MzI2LCJhdWQiOiJodHRwczovL3Rlc3Quc2VydmljZS5zaWduYWxyLm5ldDo1MDAxL2NsaWVudC8_aHViPWNoYXQiLCJuYW1laWQiOiJ1c2VyLWlkIn0.k24D5kk7KeA_JKmxaEU0gGtF4JhOlyQ2VDmITHrzPtA" ,
```
[the remark this answers is not in the document]
Comment (vwxyzh) [reply docs/internal.md]: same issue in rest-api.md

**Comment by xscript** [comment]
⏰ ⏰ ⏰  Ping...

File: docs/internal.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+After application server is started, the Azure SignalR SDK will open 5 WebSocket connections to the SignalR Service.
+The number of used connections is configurable as described at [here](./use-signalr-service.md#connectioncount).
+Messages from/to clients will be multiplexed into these connections.
+
+These connections are supposed to be connected to SignalR Service all the time.
+If a server connection is disconnected due to network issues,
+- all clients which are served by this server connection will be disconnected (read more about client and server pairing at [Transport Data between Client and Server](#transport) section);
+- the server connection will automatically start reconnecting.
+
+## Client Connections
```
Comment (vicancy) [inline docs/internal.md]: This is the implementation for ASP.NET Core SignalR, shall we point it out?

Comment (xscript) [reply docs/internal.md]: Sure. Let me add it in the title.

**Pull request merged**